stdlib/src/builtin/io.mojo

```diff
@@ -180,7 +180,12 @@ fn _snprintf[
 @no_inline
 fn _snprintf_scalar[
     type: DType
-](buffer: UnsafePointer[UInt8], size: Int, x: Scalar[type]) -> Int:
+](
+    buffer: UnsafePointer[UInt8],
+    size: Int,
+    x: Scalar[type],
+    float_format: StringLiteral = "%.17g",
```

Comment:
Here and in `_float_repr` and `_format_scalar` below: Can you please make `float_format` a parameter? Soon `StringLiteral` won't be allowed at runtime, and this would prevent some churn.